core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -159,6 +159,23 @@ public void log(String format, Object... args) {
         }
     }
 
+    /**
+     * This will log {@link RuntimeException}, if logging is enabled , and throw the runtime exception.
+     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
+     * @throws RuntimeException which is requested by this call.
+     */
+    public void logAndThrow(RuntimeException runtimeException) {
```

Comment:
I'm not sure exactly how we would want to surface the API but should there be a logAndThrowError, maybe just throwError, and logAndThrowWarning? The logging spec states that error should only be used in cases where the application won't be able to recover, such as an out of memory exception, where as warning should be used in recoverable cases, such as a REST call failing.

Reply:
Let's wait until there is a need for this API.